src/components/TradeWidget/utils.ts

```diff
@@ -19,7 +19,9 @@ export function calculateReceiveAmount(priceValue: string, sellValue: string): s
       const receiveBigNumber = sellAmount.dividedBy(price)
       // Format the "Receive at least" input amount same as PriceSuggestions price
       receiveAmount =
-        receiveBigNumber.isNaN() || !receiveBigNumber.isFinite() ? '0' : formatPriceToPrecision(receiveBigNumber)
+        receiveBigNumber.isNaN() || !receiveBigNumber.isFinite()
+          ? '0'
+          : formatPriceWithFloor(receiveBigNumber, { decimals: 18 })
```

Comment:
is it safe to pass 18 for decimals here?

Reply:
we need the max decimals from the received token, isn't it?

Reply:
hmm sorry this should be `decimalPlacesToShow` or `precision` i suppose. has nothing to do with `decimals` of token

Reply:
in `dex-js`, `formatAmount`'s `decimal` option prop isn't referring to the token decimals, it's the amount of decimal places to show:

```ts
export function formatAmount(
  params: FormatAmountParams<BN | null | undefined> | BN | null | undefined,
  _amountPrecision?: number,
): string | null {
  let amount: BN
  let precision: number

  let decimals = DEFAULT_DECIMALS <---- decimal places to show
```

Reply:
where ```DEFAULT_DECIMALS = 4`

Reply:
this was removed and fixed :)